trx: cancel/committed_version/get_address_for_key

yjh0502 opened the pull request.

yjh0502 commented on foundationdb/src/transaction.rs:
```diff
@@ -62,6 +62,37 @@ impl Transaction {
+    pub fn committed_version(&self) -> Result<i64> {
+        let trx = self.inner.inner;
+
+        let mut version = 0;
```
Done.

yjh0502 commented on foundationdb/src/future.rs:
```diff
@@ -109,6 +147,53 @@ impl FdbFutureResult {
+        }
+
+        let out_len = out_len as usize;
+        let out_strings: &[*const i8] = unsafe { std::slice::from_raw_parts(out_strings, out_len) };
```
The type should be `std::os::raw::c_char`, which maps to `i8` in (almost?) all platforms. I changed it to `c_char` for clarity.

bluejekyll commented on foundationdb/src/future.rs:
```diff
@@ -109,6 +147,53 @@ impl FdbFutureResult {
+        let mut out_keyvalues = std::ptr::null();
+        let mut out_len = 0;
+        //TODO: should return more
+        let mut more = 0;
```
Never mind on this. I read the docs more closel, we should just set a boolean.

yjh0502 replied:
I added `KeyValues<'a>` to return both `more` and '&[KeyValue]'.

bluejekyll commented:
Sorry about the conflicts. I decided to document everything today. I think we’re ready for a release to get some feedback. Also, do you want me to make you a collaborator on the repo?

yjh0502 commented:
I updated the PR to fix issues on review. I also rebased to master and fix errors from `missing_doc`. I tried to fill missing docs with matching styles. Please fill free to revise if there's a problem.

Thanks for the invite. I agree to make a the release for feedback.
